test/unit/test_discovery_v2.py

```diff
@@ -0,0 +1,1213 @@
+# Test Class for query
+#-----------------------------------------------------------------------------
+class TestQuery():
+
+    #--------------------------------------------------------
+    # Test 1: Send fake data and check response
+    #--------------------------------------------------------
+    @responses.activate
+    def test_query_response(self):
+        body = self.construct_full_body()
+        response = fake_response_QueryResponse_json
+        send_request(self, body, response)
+        assert len(responses.calls) == 1
+
+    #--------------------------------------------------------
+    # Test 2: Send only required fake data and check response
+    #--------------------------------------------------------
+    @responses.activate
+    def test_query_required_response(self):
+        # Check response with required params
+        body = self.construct_required_body()
+        response = fake_response_QueryResponse_json
+        send_request(self, body, response)
+        assert len(responses.calls) == 1
+
+    #--------------------------------------------------------
+    # Test 3: Send empty data and check response
+    #--------------------------------------------------------
+    @responses.activate
+    def test_query_empty(self):
+        check_empty_required_params(self, fake_response_QueryResponse_json)
+        check_missing_required_params(self)
+        assert len(responses.calls) == 0
+
+    #-----------
+    #- Helpers -
+    #-----------
+    def make_url(self, body):
+        endpoint = '/v2/projects/{0}/query'.format(body['project_id'])
+        url = '{0}{1}'.format(base_url, endpoint)
+        return url
+
+    def add_mock_response(self, url, response):
+        responses.add(responses.POST,
+                      url,
+                      body=json.dumps(response),
+                      status=200,
+                      content_type='application/json')
+
+    def call_service(self, body):
+        service = DiscoveryV2(authenticator=NoAuthAuthenticator(),
+                              version='2019-11-22')
+        service.set_service_url(base_url)
+        output = service.query(**body)
+        return output
+
+    def construct_full_body(self):
+        body = dict()
+        body['project_id'] = "string1"
+        body.update({
+            "collection_ids": [],
+            "filter":
+                "string1",
+            "query":
+                "string1",
+            "natural_language_query":
+                "string1",
+            "aggregation":
+                "string1",
+            "count":
+                12345,
+            "return_": [],
+            "offset":
+                12345,
+            "sort":
+                "string1",
+            "highlight":
+                True,
+            "spelling_suggestions":
+                True,
+            "table_results":
+                QueryLargeTableResults._from_dict(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['QueryLargeTableResults' may be undefined, or defined from star imports: ibm_watson.discovery_v2](https://app.codacy.com/app/watson-developer-cloud/python-sdk/pullRequest?prid=4546453)